NinjaResidence/GameChara.cpp

```diff
@@ -29,188 +29,199 @@ GameChara::~GameChara()
-		m_DisplayCharaCoordinate[0].x += 10.f;
-		m_DisplayCharaCoordinate[1].x += 10.f;
-		m_DisplayCharaCoordinate[2].x += 10.f;
-		m_DisplayCharaCoordinate[3].x += 10.f;
-		if (MapScrollX < 0) {
-			MapScrollX += 10;
-			setMapScrollX(MapScrollX);
-		}
+		m_DisplayCharaCoordinate[0].x = 820;
+		m_DisplayCharaCoordinate[1].x = 900;
+		m_DisplayCharaCoordinate[2].x = 900;
+		m_DisplayCharaCoordinate[3].x = 820;
+		m_pMapChip->m_MapScrollX -= 10;
 	}
-
-	if (m_pMapChip->getMapChipData(MapCharaPosition.Y - 1,MapCharaPositionX2) != 0 || m_pMapChip->getMapChipData(MapCharaPosition.Y - 2,MapCharaPositionX2) != 0 || m_pMapChip->getMapChipData(MapCharaPosition.Y - 3,MapCharaPositionX2) != 0 || m_pMapChip->getMapChipData(MapCharaPosition.Y - 4,MapCharaPositionX2) != 0)
+	//���Ɉړ�
+	if (m_DisplayCharaCoordinate[0].x < 300)
 	{
-		m_WorldCharaCoordinate[0].x -= 10.f;
-		m_WorldCharaCoordinate[1].x -= 10.f;
-		m_WorldCharaCoordinate[2].x -= 10.f;
-		m_WorldCharaCoordinate[3].x -= 10.f;
-		m_DisplayCharaCoordinate[0].x -= 10.f;
-		m_DisplayCharaCoordinate[1].x -= 10.f;
-		m_DisplayCharaCoordinate[2].x -= 10.f;
-		m_DisplayCharaCoordinate[3].x -= 10.f;
-		if (MapScrollX >(-30 * row)) {
-			MapScrollX -= 10;
-			setMapScrollX(MapScrollX);
-		}
+		m_DisplayCharaCoordinate[0].x = 300;
+		m_DisplayCharaCoordinate[1].x = 380;
+		m_DisplayCharaCoordinate[2].x = 380;
+		m_DisplayCharaCoordinate[3].x = 300;
+		m_pMapChip->m_MapScrollX += 10;
 	}
-	//if (GameChara::RectToRectCollisionCheak(m_WorldCharaCoordinate, m_pMapChip->CELL))
-	//{
-	//	int a = 0;
-	//}
-	//MapChara[MapCharaPositionX][MapCharaPositionY];
 }
 
+void GameChara::prevSaveMapCharaPos()
+{
+	m_prevMapCharaPositionX = (int)m_WorldCharaCoordinate[3].x;
+	m_prevMapCharaPositionX2 = (int)m_WorldCharaCoordinate[2].x;
+	m_prevMapCharaPositionY = (int)m_WorldCharaCoordinate[3].y + 10;
+}
 
-
-bool GameChara::RectToRectCollisionCheak(CUSTOMVERTEX* pObjA, CUSTOMVERTEX* pObjB)
+void GameChara::Update()
 {
-	//�l�p�`�̓����蔻��
-	if ((pObjA[0].x <= pObjB[0].x && pObjB[0].x <= pObjA[1].x) ||
-		(pObjA[0].x <= pObjB[1].x && pObjB[1].x <= pObjA[1].x) ||
-		(pObjB[0].x <= pObjA[0].x && pObjA[1].x <= pObjB[1].x))
+	GameChara::MapScroolCheck();
+	MapCharaPositionX = (int)m_WorldCharaCoordinate[3].x / 40;
+	MapCharaPositionX2 = (int)(m_WorldCharaCoordinate[2].x) / 40;
+	MapCharaPositionY = (int)(m_WorldCharaCoordinate[3].y + 10) / 40;
+	//�d�͂𖈃t���[��
+	m_WorldCharaCoordinate[0].y += 15.f;
+	m_WorldCharaCoordinate[1].y += 15.f;
+	m_WorldCharaCoordinate[2].y += 15.f;
+	m_WorldCharaCoordinate[3].y += 15.f;
+	m_DisplayCharaCoordinate[0].y += 15.f;
+	m_DisplayCharaCoordinate[1].y += 15.f;
+	m_DisplayCharaCoordinate[2].y += 15.f;
+	m_DisplayCharaCoordinate[3].y += 15.f;
+	//���̕���m���߂�
+	if (m_pMapChip->getMapChipData(MapCharaPositionY,MapCharaPositionX) != 0 || m_pMapChip->getMapChipData(MapCharaPositionY,MapCharaPositionX + 1) != 0 || m_pMapChip->getMapChipData(MapCharaPositionY,MapCharaPositionX + 2) != 0)
```

Comment:
if分の中の条件式が長すぎるような気がします
```suggestion
	if (m_pMapChip->getMapChipData(MapCharaPositionY,MapCharaPositionX)    ||
	 　m_pMapChip->getMapChipData(MapCharaPositionY,MapCharaPositionX + 1) ||
	 　m_pMapChip->getMapChipData(MapCharaPositionY,MapCharaPositionX + 2) )
```
これではだめなのですか？
負の値がVisualStudio2017ではfalseとしてとられるのでそれを意識してなら申し訳ないです
折り曲げるだけでも見やすくなるのでやってみてください

Reply:
「折り曲げる」よりも「改行する」と言いましょうｗ

Reply:
改行しました